src/d3a/models/myco_matcher/myco_matcher.py

```diff
@@ -0,0 +1,69 @@
+"""
+Copyright 2018 Grid Singularity
+This file is part of D3A.
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+from typing import Optional
+
+from d3a.d3a_core.util import is_external_matching_enabled
+from d3a.models.myco_matcher import MycoExternalMatcher, MycoInternalMatcher
+from d3a.models.myco_matcher.myco_matcher_interface import MycoMatcherInterface
+
+
+class MycoMatcher(MycoMatcherInterface):
+    """
+    Myco markets matcher, set the global matcher (Internal/External) and
+    forwards events/commands to it.
+    """
+
+    def __init__(self):
+        super().__init__()
+        self.matcher: Optional[MycoMatcherInterface] = None
+
+    def activate(self):
+        """Method to be called upon the activation of MycoMatcher."""
+        self.matcher = self.get_matcher()
+        self.matcher.activate()
+
+    @staticmethod
+    def get_matcher():
+        """Return a myco matcher instance based on the global BidOffer match type."""
+        if is_external_matching_enabled():
+            return MycoExternalMatcher()
+        else:
+            return MycoInternalMatcher()
+
+    def match_recommendations(self, **kwargs):
+        # If External matching is enabled, the trigger should be by a redis command.
+        if not is_external_matching_enabled():
+            self.matcher.match_recommendations(**kwargs)
+
+    def event_tick(self, **kwargs) -> None:
+        """Handler for the tick event."""
+        is_it_time_for_external_tick = kwargs.pop("is_it_time_for_external_tick", False)
+        if is_external_matching_enabled():
+            # If External matching is enabled, limit the number of ticks dispatched.
+            if is_it_time_for_external_tick:
+                self.matcher.event_tick(**kwargs)
+        else:
+            self.matcher.event_tick(**kwargs)
```

Comment:
```suggestion
    def event_tick(self, **kwargs) -> None:
        """Handler for the tick event."""
        self.matcher.event_tick(**kwargs)
```
Then in `MycoExternalMatcher`:
```python
from d3a.d3a_core.singletons import external_global_statistics
def event_tick(self, **kwargs):
        """Publish the tick event to the Myco client."""
        if not external_global_statistics.is_it_time_for_external_tick:
            return

        data = {"event": ExternalMatcherEventsEnum.TICK.value}
        self.myco_ext_conn.publish_json(self._events_channel, data)
```
1. You can import the singleton directly and avoid passing arguments around (in this case)
2. You don't need to check `if is_external_matching_enabled():` because, if that is `True`, you would have already changed `self.matcher` to be a `MycoExternalMatcher` instance. So you just need to change the content of `MycoExternalMatcher.event_tick()`.